wgpu-core/src/device/queue.rs

```diff
@@ -1402,9 +1405,19 @@ impl Global {
         &self,
         queue_id: QueueId,
         closure: SubmittedWorkDoneClosure,
-    ) {
+    ) -> SubmissionIndex {
+        api_log!("Queue::on_submitted_work_done {queue_id:?}");
+
+        //TODO: flush pending writes
         let queue = self.hub.queues.get(queue_id);
-        queue.on_submitted_work_done(closure);
+        let result = queue.on_submitted_work_done(closure);
+        match result {
+            Some(submission_index) => submission_index,
+            None => queue
+                .device
+                .last_successful_submission_index
+                .load(Ordering::Acquire),
```

Comment:
This should work but it might be more straight-forward to return 0 to signify that no waiting is necessary.